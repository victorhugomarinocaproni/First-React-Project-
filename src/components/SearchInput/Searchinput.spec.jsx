import { render, screen } from '@testing-library/react';

import { SearchInput } from '.';
import userEvent from '@testing-library/user-event';

describe('<SearchInput />', () => {
  it('should have a value of searchValue', () => {
    const fn = jest.fn();
    render(<SearchInput handleChange={fn} />);

    const input = screen.getByPlaceholderText(/type your search/i);
    expect(input).toBeInTheDocument();
  });

  it('should call handleChange function on each key pressed', () => {
    const fn = jest.fn();
    render(<SearchInput handleChange={fn} />);

    const input = screen.getByPlaceholderText(/type your search/i);

    const value = 'o valor';

    userEvent.type(input, value);

    expect(input.value).toBe(value);
    expect(fn).toHaveBeenCalledTimes(value.length);
  });

  it('should match the snapshot', () => {
    const fn = jest.fn();
    const { container } = render(<SearchInput handleChange={fn} />);
    expect(container).toMatchSnapshot();
  });
});
