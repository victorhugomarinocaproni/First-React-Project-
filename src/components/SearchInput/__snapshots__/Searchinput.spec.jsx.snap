// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should match the snapshot 1`] = `
<div>
  <input
    class="search-input"
    placeholder="Type your search"
    type="search"
    value=""
  />
</div>
`;
